services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/converter/ConverterUtils.java

```diff
@@ -128,55 +128,52 @@ private static int findTrimExclusiveEnd(String string, char characterToTrim, int
         }
     }
 
-    public static String[] chunk(String valueToChunk, int... splitSizes) {
-        String[] result = new String[splitSizes.length + 1];
-        int splitStartInclusive = chunkLeft(valueToChunk, result, splitSizes);
+    public static String[] chunkDateTime(String valueToChunk) {
+        Validate.isTrue(valueToChunk.contains("T") && !valueToChunk.contains("+") && !valueToChunk.contains("Z"),
+                        "Invalid sign in this format, expected [-]YYYY-MM-DDTHH:MM:SS.");
+        String negativeSig = "";
+        if (valueToChunk.substring(0, 1).equals("-")) {
+            negativeSig = "-";
+            valueToChunk = valueToChunk.substring(1);
+        }
 
-        Validate.isTrue(splitStartInclusive == valueToChunk.length(), "Value size does not match expected chunking scheme.");
+        String[] chunkedDateTime = valueToChunk.split("[-T:]");
+        chunkedDateTime[0] =  negativeSig + chunkedDateTime[0];
 
-        return result;
+        Validate.isTrue(3 * valueToChunk.length() - valueToChunk.replace("T", "").length()
+                        - valueToChunk.replace(":", "").length() - valueToChunk.replace("-", "").length() == 5
+                        && chunkedDateTime.length == 6,
+                        "Invalid DateTime (without nanoseconds) format, expected [-]YYYY-MM-DDTHH:MM:SS.");
+        return chunkedDateTime;
     }
 
-    public static String[] chunkWithRightOverflow(String valueToChunk, int... splitSizesFromLeft) {
-        String[] result = new String[splitSizesFromLeft.length + 1];
-        int splitStartInclusive = chunkLeft(valueToChunk, result, splitSizesFromLeft);
-
-        result[splitSizesFromLeft.length] = valueToChunk.substring(splitStartInclusive);
-
-        return result;
+    public static String[] chunkTime(String valueToChunk) {
+        String[] chunkedTime = valueToChunk.split(":");
+        Validate.isTrue((valueToChunk.length() - valueToChunk.replace(":", "").length()) == 2
+                         && chunkedTime.length == 3,
+                        "Invalid time format, expected HH:MM:SS.");
```

Comment:
This code is difficult to read, can we split it up into more logical and readable statements.